src/appleseed/renderer/kernel/lighting/imagebasedlighting.cpp

```diff
@@ -174,6 +229,98 @@ void compute_ibl_bsdf_sampling(
         radiance /= static_cast<float>(bsdf_sample_count);
 }
 
+void compute_ibl_bssrdf_sampling(
+    SamplingContext&        sampling_context,
+    const ShadingContext&   shading_context,
+    const EnvironmentEDF&   environment_edf,
+    const BSSRDF&           bssrdf,
+    const void*             bssrdf_data,
+    const double            bssrdf_probability,
+    const ShadingPoint&     incoming_point,
+    const ShadingPoint&     outgoing_point,
+    const Dual3d&           outgoing,
+    const double            eta,
+    const double            outgoing_fresnel,
+    const size_t            sample_count,
+    Spectrum&               radiance)
+{
+    assert(is_normalized(outgoing.get_value()));
+    assert(bssrdf_probability > 0.0);
+
+    radiance.set(0.0f);
+
+    sampling_context.split_in_place(2, sample_count);
+
+    for (size_t i = 0; i < sample_count; ++i)
+    {
+        // Generate a uniform sample in [0,1)^2.
+        const Vector2d s = sampling_context.next_vector2<2>();
+
+        // Sample the BSSRDF (hemisphere cosine).
+        Vector3d incoming = sample_hemisphere_cosine(s);
+        const double cos_in = incoming.y;
+        const double bssrdf_prob = cos_in * RcpPi * bssrdf_probability;
+        incoming = incoming_point.get_shading_basis().transform_to_parent(incoming);
+
+        // Compute Fresnel coefficient at incoming point.
+        double incoming_fresnel;
+        fresnel_transmittance_dielectric(incoming_fresnel, eta, cos_in);
+        if (incoming_fresnel <= 0.0)
+            return;
+
+        // Discard occluded samples.
+        const double transmission =
+            shading_context.get_tracer().trace(
+                incoming_point,
+                incoming,
+                VisibilityFlags::ShadowRay);
+        if (transmission == 0.0)
+            continue;
+
+        // Evaluate the diffusion profile.
+        Spectrum bssrdf_value;
+        bssrdf.evaluate(
+            bssrdf_data,
+            outgoing_point,
+            outgoing.get_value(),
+            incoming_point,
+            incoming,
+            bssrdf_value);
+
+        const double weight =
+              RcpPi
+            * cos_in
+            * transmission
+            * incoming_fresnel
+            * outgoing_fresnel;
+
+        // Evaluate the environment's EDF.
+        InputEvaluator input_evaluator(shading_context.get_texture_cache());
+        Spectrum env_value;
+        double env_prob;
+        environment_edf.evaluate(
+            shading_context,
+            input_evaluator,
+            incoming,
+            env_value,
+            env_prob);
+
+        const double mis_weight =
+            mis_power2(
+                sample_count * bssrdf_prob,
```

Comment:
Are we sure `bssrdf_prob` and `env_prob` are expressed with respect to the same measure?